### PR TITLE
cloud_lb in K3s and scoped virtual_router_id to when needed

### PR DESCRIPTION
**Pull request by larssb**
- virtual_router_id only to be used when on-prem. Won't work in the cloud.
- cloud_lb updated in scale() so that it doesn't default to true

More in the commit comments.

### CODE REVIEW

File: kvirt/cluster/k3s/__init__.py [hunk 1]
```diff
@@ -36,10 +36,12 @@ def update_ip_alias(config, nodes):
 def scale(config, plandir, cluster, overrides):
     plan = cluster
     data = {'cluster': cluster, 'kube': cluster, 'kubetype': 'k3s', 'image': 'ubuntu2004', 'sdn': 'flannel',
-            'extra_scripts': [], 'cloud_lb': True, 'cloud_native': False, 'ctlplanes': 1, 'workers': 0}
+            'extra_scripts': [], 'cloud_lb': None, 'cloud_native': False, 'ctlplanes': 1, 'workers': 0}
```
Comment (karmab) [inline kvirt/cluster/k3s/__init__.py:39]: cloud_lb should default to True

File: kvirt/cluster/k3s/__init__.py [hunk 2]
```diff
@@ -144,10 +146,12 @@ def create(config, plandir, cluster, overrides):
             else:
                 msg = "You need to define api_ip in your parameters file"
                 return {'result': 'failure', 'reason': msg}
-        if not cloud_lb and ctlplanes > 1 and data.get('virtual_router_id') is None:
-            data['virtual_router_id'] = hash(data['cluster']) % 254 + 1
+        if not cloud_lb and ctlplanes > 1:
```
[the remark this answers is not in the document]
Comment (larssb) [reply kvirt/cluster/k3s/__init__.py:149]: One win that virtual_router_id is not set unnecessarily. And if the key is not set its value is hashed generated.

File: kvirt/cluster/k3s/__init__.py (hunk 1 above)
Comment (larssb) [reply kvirt/cluster/k3s/__init__.py:39]: Can you elaborate more on why. E.g. cloud_lb is not used on-prem. So setting it to `True` won't it fault certain things if one uses `kcli scale` on-prem?

Comment (karmab) [reply kvirt/cluster/k3s/__init__.py:39]: cloud_lb is only checked when platform is cloud

Comment (larssb) [reply kvirt/cluster/k3s/__init__.py:39]: Where does the check take place. Can't find it.

Comment (karmab) [reply kvirt/cluster/k3s/__init__.py:39]: https://github.com/karmab/kcli/blob/f31deb679c11f578964750fa75fd5216f9cedbfe/kvirt/cluster/k3s/__init__.py#L215

Comment (larssb) [reply kvirt/cluster/k3s/__init__.py:39]: But, that check is in the create() not in the scale() in k3s/__init__.py ... so I still think that cloud_lb should be updated as done for the scale() in this PR.

Comment (karmab) [reply kvirt/cluster/k3s/__init__.py:39]: I think cloud_lb is not used at all in the scale process

Comment (LarsBingBong) [reply kvirt/cluster/k3s/__init__.py:39]: If it isn't then let's completely remove it from the func. definition. Agree? Then we indeed not need the extraneous code in the scale() in k3s init py.

**Comment by karmab** [comment]
where are we with this one? still needed?

**Comment by LarsBingBong** [comment]
Let's close it for now. I think, with your changes in another commit to the cloud_lb handling. We're good. The virtual_router_id could still be skipped in the cloud_lb = true scenario. But, we're good for now.

**Pull request closed without merging**